snd2png/snd2png.cpp

```diff
@@ -176,8 +176,8 @@ int main(int argc, char* argv[])
     int freqs = height / scale_factor;
     Mat grayscaleMat(height, 1024, CV_8U, Scalar(255));
 
-    if (last_non_silence - first_non_silence >= grayscaleMat.cols)
-        last_non_silence = grayscaleMat.cols - first_non_silence - 1;
+    if (last_non_silence - first_non_silence > grayscaleMat.cols)
+        last_non_silence = grayscaleMat.cols + first_non_silence;
```

Comment:
hm, the original code was introduced by @coolo in 4e08b45e in 2015. There are no tests so I guess the best way to verify is with existing openQA tests. For this I will trigger a test based on openSUSE tests using audio comparison.

Reply:
You could use this audio file (a live example from Debian):
https://openqa.debian.net/tests/325775/file/bootwalk_2:6:0-captured.wav

Then call 'snd2png my.wav my.png'

Alternatively, to enforce a segfault, you could use the 'aplay-captured.wav' file, but use e.g. audacity to add at least 1025 samples of silence at the beginning.

Reply:
git commit is now rewrapped.